File: tests/human/x/php/group_by_join.php
<?php
$customers = [
    ["id" => 1, "name" => "Alice"],
    ["id" => 2, "name" => "Bob"],
];
$orders = [
    ["id" => 100, "customerId" => 1],
    ["id" => 101, "customerId" => 1],
    ["id" => 102, "customerId" => 2],
];
$groups = [];
foreach ($orders as $o) {
    foreach ($customers as $c) {
        if ($o['customerId'] === $c['id']) {
            $groups[$c['name']][] = $o;
        }
    }
}
ksort($groups);
$stats = [];
foreach ($groups as $name => $ordersFor) {
    $stats[] = ['name'=>$name, 'count'=>count($ordersFor)];
}
_print("--- Orders per customer ---");
foreach ($stats as $s) {
    _print($s['name'], "orders:", $s['count']);
}

function _print(...$args) {
    $parts = [];
    foreach ($args as $a) {
        if (is_array($a) || is_object($a)) {
            $parts[] = json_encode($a);
        } else {
            $parts[] = strval($a);
        }
    }
    echo implode(' ', $parts), PHP_EOL;
}
?>

File: archived/tests/transpiler/x/kt/group_by_join.kt
data class GGroup(val key: Any, val items: MutableList<MutableMap<String, Int>>)
fun main() {
    val customers = mutableListOf(mutableMapOf("id" to 1, "name" to "Alice"), mutableMapOf("id" to 2, "name" to "Bob"))
    val orders = mutableListOf(mutableMapOf("id" to 100, "customerId" to 1), mutableMapOf("id" to 101, "customerId" to 1), mutableMapOf("id" to 102, "customerId" to 2))
    val stats = run {
    val _groups = mutableMapOf<Any, MutableList<MutableMap<String, Int>>>()
    for (o in orders) {
        for (c in customers) {
            if (o["customerId"] == c["id"]) {
            val _list = _groups.getOrPut(c["name"] as Any) { mutableListOf<MutableMap<String, Int>>() }
            _list.add(o)
            }
        }
    }
    val _res = mutableListOf<MutableMap<String, Any>>()
    for ((key, items) in _groups) {
        val g = GGroup(key, items)
        _res.add(mutableMapOf("name" to g.key, "count" to g.items.size))
    }
    _res
}
    println("--- Orders per customer ---")
    for (s in stats) {
        println(listOf(s["name"], "orders:", s["count"]).joinToString(" "))
    }
}
